geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/ArrowIterator.scala

```diff
@@ -56,15 +55,9 @@ object ArrowIterator {
 
   /**
     * Adapts the iterator to create simple features.
-    * WARNING - the same feature is re-used and mutated - the iterator stream should be operated on serially.
     */
-  def kvsToFeatures(): Entry[Key, Value] => SimpleFeature = {
-    val sf = new ScalaSimpleFeature(ArrowEncodedSft, "")
-    sf.setAttribute(1, GeometryUtils.zeroPoint)
-    e: Entry[Key, Value] => {
-      sf.setAttribute(0, e.getValue.get())
-      sf
-    }
+  class AccumuloArrowResultsToFeatures extends ArrowResultsToFeatures[Entry[Key, Value]] {
+    override protected def bytes(result: Entry[Key, Value]): Array[Byte] = result.getValue.get()
```

Comment:
Oh!  Having the classes lets one centralize the above code and be a little more DRY?

Reply:
To some extent yeah, but the main driver was being able to serialize them.